src/components/NavBar.vue

```diff
@@ -57,25 +57,38 @@
         </b-collapse>
     </b-navbar>
 </template>
+
 <script>
+import path from "path";
+import { rmPrefix } from "~/utils.js";
+import CONFIG from "~/../config.json";
 const REPO_URL = "https://github.com/galaxyproject/galaxy-hub/";
-const EDIT_URL = `${REPO_URL}tree/master/content/`;
+const EDIT_PATH = "tree/master/content";
 export default {
     computed: {
         editUrl() {
-            // TODO: a more robust way to do this, this only works on
-            // article-based pages and we probably don't want a hacky mess of
-            // exceptions.  For now, just default to root (which has a README,
-            // etc., so is not unreasonable) for 'special' pages.
-            let articlePath = this?.$page?.article?.fileInfo?.path;
-            if (articlePath) {
-                articlePath = articlePath.replace(/^build\//, "");
-                articlePath = articlePath.replace(/^content-md\//, "");
-                return `${EDIT_URL}${articlePath}`;
+            // This will only point to the exact Github source url for Collections which include a "fileInfo { path }"
+            // in their GraphQL query. Otherwise, (like for dynamic pages) it'll default to pointing to the repo home
+            // page (which has a README, etc., so is not unreasonable).
+            let sourcePath = getPath(this.$page);
+            if (sourcePath) {
+                sourcePath = rmPrefix(sourcePath, CONFIG.build.dirs.md);
+                return path.join(REPO_URL, EDIT_PATH, sourcePath);
```

Comment:
Probably a non-issue in reality, but we actually don't really want path.join using the platform separators -- we want explicit `/` joining these bits since it's a URL.

Reply:
touche, yeah

Reply:
(i.e. -- if you were to run this on a windows machine you would actually get `\` slashes joining the path here)

Reply:
although it does nicely handle situations where there's already a "/" prefix or suffix on one of the parts, like in `REPO_URL`